flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -498,12 +498,15 @@ protected CompletableFuture<Void> stopClusterServices(boolean cleanupHaData) {
             }
 
             if (haServices != null) {
-                try {
-                    if (cleanupHaData) {
-                        haServices.closeAndCleanupAllData();
-                    } else {
-                        haServices.close();
+                if (cleanupHaData) {
```

Comment:
Initially, I thought that it would be nicer code afterwards by just removing the if/else. But you're right with having separate try/catch blocks as well (which unfortunately, makes the code less readable).
We could introduce a default implementation {{HighAvailabilityServices#closeWithOptionalClean(boolean)}} that does this optional cleanup while closing the services. That would remove the redundant code from {{MiniCluster}} and {{ClusterEntrypoint}}.

Reply:
Thanks @XComp. Sounds great to me.
Could you help take a look again?